### PR TITLE
Unix/Linux and SteamDeck support

### PR DESCRIPTION
**Pull request by VinegarLove**
if possible, i would like the gui to be usable under linux environment.
unfortunately, i can't test the program since "stat_progression" and "itemdata" are not in the current repo.

could you give me some feedback on this?

### CODE REVIEW

**Comment by Ariescyn** [comment]
I will add those scripts to the repo, sorry. I can develop a linux version if needed. If you want to do it, you'll just have to change all of the windows commands to bash, and change some directories. The tkinter gui should work but will need a little tweaking to look good

**Comment by Ariescyn** [comment]
Also if you have any tips/advice on doing things differently please let me know!

**Comment by Ariescyn** [comment]
pyinstaller --onefile --windowed --icon=.\data\icon.ico --paths=C:\Users\heyit\dev\environments\eldenring\Lib\site-packages .\SaveManager.py .\hexedit.py .\stat_progression.py .\itemdata.py

**Comment by VinegarLove** [comment]
Summary of what I've done so far

- Generalized code for both OS using standard python function when possible
- Testing under Linux and Steam Deck
- Modified the README.md to include the instruction to run the program on Windows and Linux
- Included all the resources needed to run the code in the repo (.png, .ico, ecc)

The resources in the repo are required for the Linux / Steam Deck guys, since they'll be running the software directly from source and not with a bundled executable.

I'll test under Windows as soon as possible. If you want to go on already with the test feel free to clone my repo.

**Review by Ariescyn — COMMENTED** [review]
Does linux systems come with requests library already installed? If not then it should be added to the dependencies

**Comment by Ariescyn** [comment]
If you can, please email me at scyntacks94@gmail.com, i have a few questions

File: SaveManager.py [hunk 1]
```diff
@@ -24,20 +24,21 @@
 def get_steamid():
     """Get users steamID by pulling folder name."""
     for dir_name in os.listdir(eldenring_savedata_dir):
+        print(dir_name)
         steam_id = None
         if len(dir_name) == 17:
             steam_id = dir_name
```
[the remark this answers is not in the document]
Comment (Ariescyn) [reply SaveManager.py:35]: Sorry, a break, not return

File: os_layer.py [hunk 2]
```diff
@@ -29,7 +28,7 @@ def get_savedata_dir():
     
     return folder
 
-eldenring_savedata_dir = _get_savedata_dir()
+eldenring_savedata_dir = get_savedata_dir()
 
 def open_folder_standard_exporer(path):
     if platform.system() == "Windows":
```
Comment (Ariescyn) [inline os_layer.py:34]: When clicking 'Open file location' it couldn't find the directory

os.startfile(path.replace('/', '\\'))

Comment (VinegarLove) [reply os_layer.py:35]: fixed the missing break

**Comment by VinegarLove** [comment]
> Does linux systems come with requests library already installed? If not then it should be added to the dependencies

You're right. I've added requests to the dependencies

**Pull request merged**